slick/src/sphinx/developers.rst

```diff
@@ -0,0 +1,155 @@
+For Slick developers
+======================
+
```

Comment:
I'd be tempted to put some intro words before the links. Something like:

> Getting started with working on the Slick codebase is easy.
> All you need is Java 8, git and sbt 0.13. 
> The Slick source code has all embedded databases preconfigured for unit tests, so you can run a comprehensive test suite without any further setup.

...and we can link to a community process page, once we have such a page (after this PR, I suspect).